### PR TITLE
Pixel Ntuple Fitters on cpu

### PR DESCRIPTION
**Pull request by VinInn**
this PR (built atop of #266) consolidates pixel "Ntuple" fitting on CPU.

   * the mixed workflow (only fit on GPU) has been removed
   * code duplication avoided
   * new naming
   * Riemann vs BrokenLine  selected using boolean

more cleanup and renaming needed for sure

new workflows
```
runTheMatrix.py -n | grep 10824.5
10824.5 TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_pixelTrackingOnly_2018+HARVESTFull_pixelTrackingOnly_2018
10824.51 TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_pixelTrackingOnlyNtupleFit_2018+HARVESTFull_pixelTrackingOnlyNtupleFit_2018
10824.52 TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_pixelTrackingOnlyGPU_2018+HARVESTFull_pixelTrackingOnlyGPU_2018
```

```
runTheMatrix.py --list=10824.5 -j 0
runTheMatrix.py --list=10824.51 -j 0
runTheMatrix.py --list=10824.52 -j 0
```
all succeed

MTV (from modified 10824.51)
(comparing Riemann vs BrokenLine on GPU and CPU) in
http://innocent.home.cern.ch/innocent/RelVal/pixOnly105FitsonCPU/

ps  BrokenLine is default on both CPU and GPU

### CODE REVIEW

File: RecoPixelVertexing/PixelTrackFitting/plugins/PixelNtupletsFitterProducer.cc [hunk 1]
```diff
@@ -11,43 +11,41 @@
...
   }
-  ~PixelFitterByRiemannParaboloidProducer() override {}
+  ~PixelNtupletsFitterProducer() override {}
 
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
     edm::ParameterSetDescription desc;
-    desc.add<bool>("useErrors", true);
-    desc.add<bool>("useMultipleScattering", true);
-    descriptions.add("pixelFitterByRiemannParaboloidDefault", desc);
+    desc.add<bool>("useRiemannFit", false);
```
Comment (makortel) [inline RecoPixelVertexing/PixelTrackFitting/plugins/PixelNtupletsFitterProducer.cc]: The configuration would be more self-documenting if we'd use a string here, but that can be done later as well.

Comment (VinInn) [reply RecoPixelVertexing/PixelTrackFitting/plugins/PixelNtupletsFitterProducer.cc]: yes sure, as you suggested in #265

Comment (makortel) [reply RecoPixelVertexing/PixelTrackFitting/plugins/PixelNtupletsFitterProducer.cc]: I had more in mind something along `desc.add<std::string>("fitter", "BrokenLine");` with valid values being e.g. (for now) "BrokenLine", "RiemannFit", but it's not really critical (and may be a matter of taste).

**Comment by felicepantaleo** [comment]
:construction: Validation running at vinavx2.cern.ch:/data/user/fpantale/patatrack-validation/run.clDGiWoXrq ...

**Comment by felicepantaleo** [comment]
:construction: Validation running at vinavx2.cern.ch:/data/user/fpantale/patatrack-validation/run.telHQ5Fjfk ...

**Comment by felicepantaleo** [comment]
:construction: Validation running at vinavx2.cern.ch:/data/user/fpantale/patatrack-validation/run.orc7nPYTy4 ...

**Comment by felicepantaleo** [comment]
:construction: Validation running at vinavx2.cern.ch:/data/user/fpantale/patatrack-validation/run.pAo1AIUtM2 ...

**Comment by fwyzard** [comment]
@VinInn please rebase this PR on top of CMSSW_10_6_0_pre2 and factor it into independent changes

**Pull request closed without merging**